easybuild/easyblocks/generic/cmakemake.py

```diff
@@ -317,9 +317,60 @@ def configure_step(self, srcdir=None, builddir=None):
                 self.cfg['configopts']])
 
         (out, _) = run_cmd(command, log_all=True, simple=False)
+        self.check_python_paths()
 
         return out
 
+    def check_python_paths(self):
+        """Check that there are no detected Python paths outside the EB installed PythonF"""
+        if not os.path.exists('CMakeCache.txt'):
+            self.log.warning("CMakeCache.txt not found. Python paths checks skipped.")
+            return
+        cmake_cache = read_file('CMakeCache.txt')
+        if not cmake_cache:
+            self.log.warning("CMake Cache could not be read. Python paths checks skipped.")
+            return
+
+        self.log.info("Checking Python paths")
+
+        python_paths = {
+            "executable": [],
+            "include_dir": [],
+            "library": [],
+        }
+        PYTHON_RE = re.compile(r"_?(?:Python|PYTHON)\d_(EXECUTABLE|INCLUDE_DIR|LIBRARY)[^=]*=(.*)")
+        for line in cmake_cache.splitlines():
+            match = PYTHON_RE.match(line)
+            if match:
+                path_type = match[1].lower()
+                path = match[2].strip()
+                self.log.info(f"Python {path_type} path: {path}")
+                python_paths[path_type].append(path)
+
+        ebrootpython_path = get_software_root("Python")
+        if not ebrootpython_path:
+            if any(python_paths.values()) and not self.toolchain.comp_family() == toolchain.SYSTEM:
+                self.log.warning("Found Python paths in CMake cache but Python is not a dependency")
+            # Can't do the check
+            return
+        ebrootpython_path = os.path.realpath(ebrootpython_path)
+
+        errors = []
+        for path_type, paths in python_paths.items():
+            for path in paths:
+                if path.endswith('-NOTFOUND'):
+                    continue
+                if not os.path.exists(path):
+                    errors.append(f"Python {path_type} does not exist: {path}")
```

Comment:
Since this is a PR to `develop`, we need to be careful with f-strings, since EasyBuild 4.x is still supposed to support Python 2.7:
```suggestion
                    errors.append("Python %s does not exist: %s" % (path_type, path))
```